codalab/worker_manager/worker_manager.py

```diff
@@ -46,6 +46,16 @@ class WorkerManager(object):
       burden of deciding on the user.
     """
 
+    # Subcommand name to use for this worker manager type
+    NAME = 'worker-manager'
+
+    @staticmethod
+    def add_parser(subparsers):
+        """
+        Add any arguments specific to this worker manager to the given subparser
```

Comment:
It might be more user-friendly for the sub-classes to have a `add_arguments_to_subparser(subparser)` function, and then create the parser on the outside (which might help with sharing code).

Reply:
So the actual sub-command is created by the `main` script and each sub-class just adds its arguments to this sub-parser, rather than creating the sub-parser in the sub-class as well?

Reply:
yes